sentry/src/routes/campaign.rs

```diff
@@ -25,32 +68,197 @@ pub async fn create_campaign<A: Adapter>(
 
     // TODO AIP#61: Validate Campaign
 
-    let error_response = ResponseError::BadRequest("err occurred; please try again later".into());
+    let error_response = ResponseError::BadRequest("err occurred; please try again later".to_string());
 
-    // insert Campaign
+    let accounting_spent = get_accounting_spent(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+
+    // TODO: AIP#61: Update when changes to Spendable are ready
+    let latest_spendable = fetch_spendable(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+    let remaining_for_channel = get_total_remaining_for_channel(&accounting_spent, &latest_spendable).ok_or(ResponseError::BadRequest("couldn't get total remaining for channel".to_string()))?;
 
-    // match insert_campaign(&app.pool, &campaign).await {
-    //     Err(error) => {
-    //         error!(&app.logger, "{}", &error; "module" => "create_channel");
-
-    //         match error {
-    //             PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
-    //                 Err(ResponseError::Conflict(
-    //                     "channel already exists".to_string(),
-    //                 ))
-    //             }
-    //             _ => Err(error_response),
-    //         }
-    //     }
-    //     Ok(false) => Err(error_response),
-    //     _ => Ok(()),
-    // }?;
-
-    let create_response = SuccessResponse { success: true };
+    if campaign.budget > remaining_for_channel {
+        return Err(ResponseError::BadRequest("Not Enough budget for campaign".to_string()));
+    }
+
+    // If the channel is being created, the amount spent is 0, therefore remaining = budget
+    increase_remaining_for_campaign(&app.redis.clone(), campaign.id, campaign.budget).await.map_err(|_| ResponseError::BadRequest("Couldn't update remaining while creating campaign".to_string()))?;
+
+    // insert Campaign
+    match insert_campaign(&app.pool, &campaign).await {
+        Err(error) => {
+            error!(&app.logger, "{}", &error; "module" => "create_campaign");
+            match error {
+                PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
+                    Err(ResponseError::Conflict(
+                        "Campaign already exists".to_string(),
+                    ))
+                }
+                _ => Err(error_response),
+            }
+        }
+        Ok(false) => Err(ResponseError::BadRequest("Encountered error while creating Campaign; please try again".to_string())),
+        _ => Ok(()),
+    }?;
 
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
 
+
+// tested
```

Comment:
not entirely sure if we need a whole function for this.
It only performs single line `latest_spendable.deposit.total.checked_sub(accounting_spent)`